From 0581b72b9181808af4bd2fab05c90e574077262a Mon Sep 17 00:00:00 2001
From: Grzegorz Jaszczyk <jaz@semihalf.com>
Date: Thu, 11 Aug 2016 10:30:49 +0200
Subject: [PATCH 0449/1345] arm64: dts: marvell: add pinctrl for Armada 8K CP0
 and for CP1

commit  5ce9b757f3d64994a201163f3dc3a85311582827 from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

The base addresses for pinctrl are the same for Armada 8K and Armada 7K but
different compatible strings need to be used. Therefore introduce the
pinctrl node in the lowest dtsi layer - armada-cp110-1.dtsi and
armada-cp110.dtsi, but define compatible property in upper dtsi layers.

Also it is worth recalling once again that Armada 8K contains two CP110
blocks (CP0 and CP1), while Armada 7K contains only one CP110 block.

Change-Id: Id3a99f04da8403950f8ce257234d760c28201624
Signed-off-by: Grzegorz Jaszczyk <jaz@semihalf.com>
Signed-off-by: Igal Liberman <igall@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/32375
Reviewed-by: Omri Itach <omrii@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Tested-by: Star_Automation <star@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/marvell/armada-7040.dtsi    |    2 +
 arch/arm64/boot/dts/marvell/armada-8040.dtsi    |   57 +++++++++++++++++++++++
 arch/arm64/boot/dts/marvell/armada-cp110-1.dtsi |    5 ++
 arch/arm64/boot/dts/marvell/armada-cp110.dtsi   |    3 +-
 4 files changed, 65 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/marvell/armada-7040.dtsi b/arch/arm64/boot/dts/marvell/armada-7040.dtsi
index 50570d5..9cf4002 100644
--- a/arch/arm64/boot/dts/marvell/armada-7040.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-7040.dtsi
@@ -69,6 +69,8 @@
 	cp110-master {
 		config-space@f2000000 {
 			pinctrl@440000 {
+			compatible = "marvell,a70x0-pinctrl";
+
 				i2c0_pins: i2c-pins-0 {
 					marvell,pins = "mpp37", "mpp38";
 					marvell,function = "i2c0";
diff --git a/arch/arm64/boot/dts/marvell/armada-8040.dtsi b/arch/arm64/boot/dts/marvell/armada-8040.dtsi
index 9e07cf0..9690c025 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-8040.dtsi
@@ -133,6 +133,34 @@
 			ppv22@000000 {
 				dma-coherent;
 			};
+			pinctrl@440000 {
+				compatible = "marvell,a80x0-cp0-pinctrl";
+				reg = <0x440000 0x20>;
+				cpm_i2c0_pins: i2c-pins-0 {
+					marvell,pins = "mpp37", "mpp38";
+					marvell,function = "i2c0";
+				};
+				cpm_i2c1_pins: i2c-pins-1 {
+					marvell,pins = "mpp35", "mpp36";
+					marvell,function = "i2c1";
+				};
+				cpm_ge2_rgmii_pins: ge-rgmii-pins-0 {
+					marvell,pins = "mpp44", "mpp45", "mpp46",
+						       "mpp47", "mpp48", "mpp49",
+						       "mpp50", "mpp51", "mpp52",
+						       "mpp53", "mpp54", "mpp55";
+					marvell,function = "ge1";
+				};
+				pca0_pins: pca0_pins {
+					marvell,pins = "mpp62";
+					marvell,function = "gpio";
+				};
+				cpm_sdhci_pins: cpm-sdhi-pins {
+					marvell,pins = "mpp56", "mpp57", "mpp58",
+						       "mpp59", "mpp60", "mpp61";
+					marvell,function = "sdio";
+				};
+			};
 			usb3h0: usb3@500000 {
 				#stream-id-cells = <1>;
 			};
@@ -174,6 +202,35 @@
 			ppv22@000000 {
 				dma-coherent;
 			};
+			pinctrl@440000 {
+				compatible = "marvell,a80x0-cp1-pinctrl";
+				reg = <0x440000 0x10>;
+				cps_ge1_rgmii_pins: ge-rgmii-pins-0 {
+					marvell,pins = "mpp0", "mpp1", "mpp2",
+						       "mpp3", "mpp4", "mpp5",
+						       "mpp6", "mpp7", "mpp8",
+						       "mpp9", "mpp10", "mpp11";
+					marvell,function = "ge0";
+				};
+				cps_tdm_pins: tdm-pins {
+					marvell,pins = "mpp0", "mpp1", "mpp2",
+						       "mpp3", "mpp4", "mpp5";
+					marvell,function = "tdm";
+				};
+				cps_spi0_pins: spi-pins-0 {
+					marvell,pins = "mpp7", "mpp8", "mpp9",
+						       "mpp10", "mpp11";
+					marvell,function = "spi0";
+				};
+				cps_spi1_pins: spi-pins-1 {
+					marvell,pins = "mpp13", "mpp14", "mpp15",
+						       "mpp16";
+					marvell,function = "spi1";
+				};
+			};
+			rtc@284000 {
+				status = "okay";
+			};
 		};
 		pcie@0x600000 {
 			msi-parent = <&gic_v2m0>;
diff --git a/arch/arm64/boot/dts/marvell/armada-cp110-1.dtsi b/arch/arm64/boot/dts/marvell/armada-cp110-1.dtsi
index 2702f08..a170533 100644
--- a/arch/arm64/boot/dts/marvell/armada-cp110-1.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-cp110-1.dtsi
@@ -37,6 +37,11 @@ gateclk_1: clock-gating-control@440220 {
 	#clock-cells = <1>;
 };
 */
+
+pinctrl@440000 {
+	reg = <0x440000 0x10>;
+};
+
 sata@540000 {
 	compatible = "marvell,armada-3700-ahci";
 	reg = <0x540000 0x30000>;
diff --git a/arch/arm64/boot/dts/marvell/armada-cp110.dtsi b/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
index 6561083..ce1d60b 100644
--- a/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
@@ -37,8 +37,7 @@ gateclk: clock-gating-control@440220 {
 };
 
 pinctrl@440000 {
-	compatible = "marvell,a70x0-pinctrl";
-	reg = <0x440000 0x10>;
+	reg = <0x440000 0x20>;
 };
 
 gpio@440100 {
-- 
1.7.9.5

